### ISSUE TITLE
Support input data that don't fit in memory

### ISSUE THREAD

**Issue by nanxstats** [nanxstats/tinytopics#23]
When `X` is really large, it might not fit into the CPU RAM.

We might want to extend `fit_model()` to support using `Dataset` and `DataLoader` with custom loading (from disk), by subclassing `torch.utils.data.Dataset` and implement the `__len__` and `__getitem__` methods.

Need an article to document this feature.

**Comment by nanxstats** [comment]
Mark as not doing for now as it might involve updating the scheduler with fractional epochs which is less than ideal.
